domain/secret/service/service.go

```diff
@@ -137,33 +109,58 @@ func (s *SecretService) getBackendForUserSecrets(ctx context.Context, accessor S
 	}
 	backend, err := s.getBackend(&cfg)
 	if err != nil {
-		return nil, "", errors.Trace(err)
+		return nil, "", jujuerrors.Trace(err)
 	}
 	return backend, activeBackendID, nil
 }
 
 func (s *SecretService) loadBackendInfo(ctx context.Context, activeOnly bool) error {
-	s.backends = make(map[string]provider.SecretsBackend)
-	info, err := s.adminConfigGetter(ctx)
+	mUUID, err := s.secretState.GetModelUUID(ctx)
 	if err != nil {
-		return errors.Trace(err)
+		return errors.Errorf("getting model UUID: %w", err)
 	}
-	s.activeBackendID = info.ActiveID
-	for id, cfg := range info.Configs {
-		if id != info.ActiveID && activeOnly {
+	modelUUID := coremodel.UUID(mUUID)
+
+	modelBackend, err := s.secretBackendState.GetModelSecretBackendDetails(ctx, modelUUID)
+	if err != nil {
+		return errors.Errorf("getting model secret backend: %w", err)
+	}
+	s.activeBackendID = modelBackend.SecretBackendID
+
+	backends, err := s.secretBackendState.ListSecretBackendsForModel(ctx, modelUUID, true)
+	if err != nil {
+		return errors.Errorf("listing secret backends: %w", err)
+	}
+
+	s.backends = make(map[string]provider.SecretsBackend)
+	for _, b := range backends {
+		if activeOnly && b.ID != s.activeBackendID {
 			continue
 		}
-		s.backends[id], err = s.getBackend(&cfg)
+
+		cfg := provider.ModelBackendConfig{
+			ControllerUUID: modelBackend.ControllerUUID,
+			ModelUUID:      mUUID,
+			ModelName:      modelBackend.ModelName,
+			BackendConfig: provider.BackendConfig{
+				BackendType: b.BackendType,
+				Config:      b.Config,
+			},
+		}
+
+		s.backends[b.ID], err = s.getBackend(&cfg)
 		if err != nil {
-			if id != info.ActiveID && cfg.BackendType == kubernetes.BackendType {
+			if b.ID != s.activeBackendID && cfg.BackendType == kubernetes.BackendType {
 				// TODO(secrets) - on an iaas controller, attempting to get the "model" k8s backend fails
```

Comment:
I'd like to see us not having these hang around long term.

Reply:
This can actually be removed now. It was fixed but the todo wasn't taken out.